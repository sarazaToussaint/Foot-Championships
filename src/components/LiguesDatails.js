import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BiRightArrowCircle } from 'react-icons/bi';
import { useLocation } from 'react-router-dom';
// import { FaMicrophone } from 'react-icons/fa';
// import { IoChevronBackOutline, IoSettingsOutline } from 'react-icons/io5';
import { fetchDetails } from './redux/details';

const LiguesDatails = () => {
  const details = useSelector((state) => state.details);
  const dispatch = useDispatch();
  const location = useLocation();
  const { id } = location.state;

  useEffect(() => {
    dispatch(fetchDetails(id));
  }, []);

  return (
    <>
      {/* <div className="nav-container">
        <Link to="/"><IoChevronBackOutline style={{ color: '#fff' }} /></Link>
        <p>title</p>
        <div className="nav-icons">
          <FaMicrophone />
          <IoSettingsOutline />
        </div>
      </div> */}
      <div className="grid-container">
        <div>
          <h2>Check all Club info in this Championship</h2>
        </div>
        <div>
          Total teams:
          {' '}
          {details.length}
        </div>
        {details.map((club) => (
          <div key={club.name} className="club-container">
            <div className="name-logo">
              <img src={club.logo} alt={club.name} />
              <p>{club.name}</p>
            </div>
            <div className="points-arrow">
              <p>
                {club.rank}
                {' '}
                points
              </p>
              <BiRightArrowCircle />
            </div>
          </div>
        ))}
      </div>

    </>
  );
};

export default LiguesDatails;
