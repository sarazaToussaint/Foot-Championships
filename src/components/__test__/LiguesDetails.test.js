import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configStore';
import LiguesDatails from '../LiguesDatails';

describe('Leagues test', () => {
  it('renders leagues', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <LiguesDatails />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
