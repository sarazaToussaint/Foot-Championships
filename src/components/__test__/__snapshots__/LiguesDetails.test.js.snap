// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leagues test renders leagues 1`] = `
<div
  className="Container"
>
  <div
    className="totalClub"
  >
    Total teams:
     
    0
  </div>
</div>
`;
